### PR TITLE
Replace curl with guzzle

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
Replace curl with guzzle

Fixes: https://app.asana.com/0/564264490825835/1137562463503385

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have created or modified unit tests to sufficiently cover my changes.

### CODE REVIEW

File: lib/Boltpay/Bugsnag/Notification.php [hunk 1]
```diff
@@ -171,16 +171,11 @@ public function postJSON($url, $data)
...
-        // cURL supports both timeouts and proxies
-
         try {
-            if (function_exists('curl_version')) {
-                $this->postWithCurl($url, $body);
-            } elseif (ini_get('allow_url_fopen')) {
-                $this->postWithFopen($url, $body);
-            } else {
-                error_log('Bugsnag Warning: Couldn\'t notify (neither cURL or allow_url_fopen are available on your PHP installation)');
+            if (method_exists('GuzzleHttp\Client','sendAsync')) {
```
Comment (matt-thomason) [inline lib/Boltpay/Bugsnag/Notification.php]: Why check for `sendAsync`?

File: app/code/community/Bolt/Boltpay/controllers/ConfigurationController.php [hunk 2]
```diff
@@ -116,33 +116,32 @@ protected function checkPublishableKeyMultiPage()
...
+            'X-Publishable-Key'=> $key,
         );
-        curl_setopt($ch, CURLOPT_HTTPHEADER, $headerInfo);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($ch, CURLOPT_HEADER, true);
 
-        curl_exec($ch);
-        $response = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-        curl_close($ch);
+        try{
```
[the remark this answers is not in the document]
Comment (johnmccombs1) [reply app/code/community/Bolt/Boltpay/controllers/ConfigurationController.php]: @mccottry @matt-thomason this has been fixed. Can you review?

File: lib/Boltpay/Bugsnag/Client.php [hunk 3]
```diff
@@ -363,22 +363,6 @@ public function setProxySettings(array $proxySettings)
...
-     * Set custom curl options.
-     *
-     * @param array $curlOptions an array with curl options. Eg:
-     *        array(
-     *            CURLOPT_IPRESOLVE => CURL_IPRESOLVE_V4
-     *        )
-     *
-     * @return $this
-     */
-    public function setCurlOptions(array $curlOptions)
```
[the remark this answers is not in the document]
Comment (johnmccombs1) [reply lib/Boltpay/Bugsnag/Client.php]: @matt-thomason this is the only reference that I can find.

File: app/code/community/Bolt/Boltpay/Helper/ApiTrait.php [hunk 4]
```diff
@@ -118,85 +118,74 @@ public function verify_hook($payload, $hmacHeader)
...
+     * @param $command
+     * @param $data
+     * @param string $object
+     * @param string $type
      * @param null $storeId
-     * @throws  Mage_Core_Exception  thrown if an error is detected in a response
-     * @return mixed           Object derived from Json got as a response
+     * @return mixed
+     * @throws Mage_Core_Exception
+     * @throws \GuzzleHttp\Exception\GuzzleException
```
Comment (mccottry) [inline app/code/community/Bolt/Boltpay/Helper/ApiTrait.php]: The comments here have become worse then the original.  Please re-apply them.

File: app/code/community/Bolt/Boltpay/Helper/ApiTrait.php [hunk 5]
```diff
@@ -71,31 +82,20 @@ private function verify_hook_api($payload, $hmacHeader)
...
-
-            $response = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-            $this->setCurlResultWithHeader($ch, $result);
-
-            $resultJSON = $this->getCurlJSONBody();
+            $resultJSON = (string)$this->getApiClient()->post($url,$payload,$httpHeader)->getBody();
             $this->addMetaData(array('BOLT API RESPONSE' => array('verify-hook-api-response'=>$resultJSON)),true);
 
-            return $response == 200;
+            return $resultJSON;
```
Comment (mccottry) [inline app/code/community/Bolt/Boltpay/Helper/ApiTrait.php]: Similar to before, I'm very uneasy with changing the logic here from checking for a 200 response code to checking for the existence of a response body.  We should continue to check the HTTP response code here.

**Pull request merged**